scripts/quests/otherAreas/An_Affable_Adamantking.lua

```diff
@@ -0,0 +1,321 @@
+                    elseif quest:getVar(player, 'Prog') == 5 then
+                        return quest:progressEvent(63) -- congratulate player on fight
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, { xi.item.QUADAV_BARBUT }) and
+                        quest:getVar(player, 'Prog') == 6
+                    then
+                        return quest:progressEvent(64) -- reward player with Da'Vhu's Barbut
+                    end
+                end,
+            },
+
+            ['Diamond_Quadav'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET == mob:getID() then
+                        for i = qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 1, qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 6 do
+                            DespawnMob(i)
+                        end
+
+                        player:delKeyItem(xi.ki.ORCISH_SEEKER_BATS)
+                        GetNPCByID(qulunDomeID.npc.BEASTMENS_BANNER):setLocalVar('engaged', os.time()) -- no time gate after death
+                        mob:setLocalVar('died', 1) -- set died so despawn doesn't trigger 3 min timer
+                        quest:setVar(player, 'Prog', 5)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [61] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.ORCISH_SEEKER_BATS)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [62] = function(player, csid, option, npc)
+                    if
+                        npcUtil.popFromQM(player, npc, {
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 1,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 2,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 3,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 4,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 5,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 6,
+                        }, { claim = true, hide = 0 })
+                    then
+                        GetNPCByID(qulunDomeID.npc.BEASTMENS_BANNER):setLocalVar('engaged', 1)
+                    end
+                end,
+
+                [63] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 6)
+                end,
+
+                [64] = function(player, csid, option, npc)
+                    if player:getFreeSlotsCount() > 0 then
+                        player:confirmTrade()
+                        npcUtil.giveItem(player, xi.item.DAVHUS_BARBUT)
```

Comment:
This reward can be covered by the reward table, and then check if quest:complete(player).  This will also help to ensure that the player successfully received the item prior to setting other values.

Reply:
Moved reward to reward table and nested all prog/var setting into `quest:complete(player)` check